doc/docs/Installation.md

```diff
@@ -270,11 +270,146 @@ By default, Meep's configure script tries to guess the gcc `-march` flag for the
+export LDFLAGS="-L/usr/local/lib -L/usr/lib/x86_64-linux-gnu/hdf5/serial"
+export CPPFLAGS="-I/usr/local/include -I/usr/include/hdf5/serial"
+export PYTHONPATH="$HOME/install/meep/python"
+export GUILE_WARN_DEPRECATED="no"
+
+sudo apt-get update
+sudo apt-get -y dist-upgrade
+sudo apt-get -y install libblas-dev liblapack-dev libgmp-dev libunistring-dev libmatheval-dev swig libgsl-dev libatomic-ops-dev libgc-dev libffi-dev libltdl-dev autoconf pkg-config libpng16-dev git guile-2.0-dev libhdf5-dev libfftw3-dev libpython3.5-dev python3-numpy python3-h5py
+
+mkdir ~/install
+
+cd ~/install
+git clone https://github.com/stevengj/harminv.git
+cd harminv/
+sh autogen.sh --enable-shared
+make && sudo make install
+
+cd ~/install
+git clone https://github.com/stevengj/libctl.git
+cd libctl/
+sh autogen.sh --enable-shared
+make && sudo make install
+
+cd ~/install
+git clone https://github.com/stevengj/h5utils.git
+cd h5utils/
+sh autogen.sh
+make distclean
+./configure --enable-shared
+make && sudo make install
+
+cd ~/install
+git clone https://github.com/stevengj/mpb.git
+cd mpb/
+sh autogen.sh --enable-shared
+make && make check && sudo make install
+make distclean
+sh autogen.sh --enable-shared --with-inv-symmetry
+make && make check && sudo make install
+
+cd ~/install
+git clone https://github.com/stevengj/meep.git
+cd meep/
+sh autogen.sh
+make distclean
+./configure --enable-shared --enable-maintainer-mode --with-python
+make && make check && sudo make install
+```
+
+#### Parallel
+
+```bash
+#!/bin/bash
+
+export LD_LIBRARY_PATH="/usr/local/lib:/usr/local/lib/openmpi:${HOME}/install/lib"
+export LDFLAGS="-L/usr/local/lib -L/usr/local/lib/openmpi -L${HOME}/install/lib"
+export CPPFLAGS="-I/usr/local/include -I/usr/local/include/openmpi -I${HOME}/install/include"
+export PATH="${PATH}:${HOME}/install/bin"
+export PYTHONPATH="${HOME}/install/meep/python"
+export GUILE_WARN_DEPRECATED="no"
+
+sudo apt-get update
+sudo apt-get -y dist-upgrade
+sudo apt-get -y install libblas-dev liblapack-dev libgmp-dev libunistring-dev libmatheval-dev swig libgsl-dev libatomic-ops-dev libgc-dev libffi-dev libltdl-dev autoconf pkg-config libpng16-dev git guile-2.0-dev libpython3.5-dev python3-numpy python3-pip
+
+mkdir ~/install
+
+cd ~/install
+git clone https://github.com/stevengj/harminv.git
+cd harminv/
+sh autogen.sh --enable-shared
+make && sudo make install
+
+cd ~/install
+git clone https://github.com/stevengj/libctl.git
+cd libctl/
+sh autogen.sh --enable-shared
+make && sudo make install
+
+cd ~/install
+wget --no-check-certificate https://www.open-mpi.org/software/ompi/v2.0/downloads/openmpi-2.0.4.tar.gz
+tar xvzf openmpi-2.0.4.tar.gz
+cd openmpi-2.0.4/
+./configure --enable-shared
+make && make check && sudo make install
+
+export CC=mpicc
+export CXX=mpic++
+
+cd ~/install
+wget --no-check-certificate https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.1/src/hdf5-1.10.1.tar.gz
+tar xvzf hdf5-1.10.1.tar.gz
```

Comment:
Why can't we use Ubuntu's `libhdf5-openmpi-dev` package?

Note that you can still build the serial versions of Meep and MPB, you just need to build with `CC=mpicc CXX=mpicxx`.

Reply:
Building parallel `hdf5` from source after setting the environment variables `export CC=mpicc` and `export CC=mpicc` ensures that the correct OpenMPI compilers are used throughout the entire build. In previous tests, I found that using Ubuntu's `libhdf5-openmpi-dev` package often lead to linking errors when building Meep (i.e., the `hdf5` libraries could not be found, etc.).

Reply:
Use `CC=mpicc.openmpi` and `CXX=mpic++.openmpi` in case the user installed other MPI versions too.

Reply:
This [install script](http://ab-initio.mit.edu/~oskooi/meep_discuss/build_meep_python_mpi2.sh) uses `libhdf5-openmpi-dev` and builds successfully. The only other packages built from source are: `harminv`, `libctl`, `h5utils`, and `mpb`. 

There are two minor issues: (1) `h5ls` is missing, and (2) MPI jobs produce the following warning:

```
--------------------------------------------------------------------------
[[47997,1],0]: A high-performance Open MPI point-to-point messaging module
was unable to find any relevant network interfaces:

Module: OpenFabrics (openib)
  Host: ip-XXX-XX-XX-XXX

Another transport will be used instead, although this may result in
lower performance.
--------------------------------------------------------------------------
```
This can be circumvented by [adding a flag to prevent searching for Infiniband hardware](http://www.olmjo.com/blog/2012/09/30/openmpi-warning/). 

All that is left now is to omit the environment variables using `-rpath`...

Reply:
`h5ls` is in the `hdf5-tools` Ubuntu package — see the [Ubuntu package search](https://packages.ubuntu.com/search?searchon=contents&keywords=h5ls&mode=exactfilename&suite=artful&arch=any).